include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -15,63 +15,79 @@ namespace impl
     public:
         IAsyncQueueDispatcherBase() = default;
         ~IAsyncQueueDispatcherBase() = default;
+        // dont want to play around with relaxed memory ordering yet
         struct request_base_t
         {
+                enum E_STATE : uint32_t
+                {
+                    ES_INITIAL=0,
+                    ES_RECORDING=1,
+                    ES_PENDING=2,
+                    ES_EXECUTING=3,
```

Comment:
whats the difference between PENDING and EXECUTING state?

Reply:
ah, ok, recorded, but not yet being executed

Reply:
PENDING is the state between the end of writing the request struct into the circular buffer and the start of execution of the request (between the two user defined CRTP static polymorphic methods)

EXECUTING is the state during the call to the user defined CRTP static polymorphic request execution routine

Reply:
Originally I just had pending state for both cases, but the Cancellable requests required that I make a distinction between pending (which can be cancelled) and executing (which must not)